src/infrastructure/channels/mod.rs

```diff
@@ -0,0 +1,24 @@
+///! This module provides channels for processing packets of different reliabilities.
+
+mod unreliable_channel;
+mod sequenced_channel;
+mod reliable_channel;
+
+use infrastructure::DeliveryMethod;
+use error::NetworkResult;
+use packet::PacketData;
+use std::io::Cursor;
+
+pub use self::unreliable_channel::UnreliableChannel;
+pub use self::sequenced_channel::SequencedChannel;
+pub use self::reliable_channel::ReliableChannel;
+
+/// This provides an abstraction for processing packets to their given reliability.
+pub trait Channel {
```

Comment:
I would like to have some better names for these trait functions, I can try and brainstorm some.